WORKSPACE

```diff
@@ -1,27 +1,32 @@
 # gazelle:repository_macro hack/build/repos.bzl%go_repositories
 workspace(name = "com_github_jetstack_cert_manager")
 
+load("//:workspace.bzl", "check_min_bazel_version")
+
+# rules_go v0.28.0 requires Bazel v4.0.0 as minimum
+# https://github.com/bazelbuild/rules_go/releases/tag/v0.28.0
+check_min_bazel_version("4.0.0")
```

Comment:
I uninstalled bazel3 and installed bazel4 
```
sudo dnf remove bazel3
sudo dnf install bazel4
...
sudo dnf install bazel4
Last metadata expiration check: 1:02:01 ago on Thu 30 Sep 2021 11:55:52 BST.
Dependencies resolved.
==============================================================================================================================================================================================================================================
 Package                                                 Architecture                             Version                                                 Repository                                                                     Size
==============================================================================================================================================================================================================================================
Installing:
 bazel4                                                  x86_64                                   4.2.1-0.fc34                                            copr:copr.fedorainfracloud.org:vbatts:bazel                                    27 M
...
```


And then reran the tests. 
This time they ran, but failed

```sh
bazel test //pkg/util/...
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
INFO: SHA256 (https://golang.org/dl/?mode=json&include=all) = 47299df6f0166ee1658fcd04d7e5db43a6b537d18f7886121ad7de9598371be1
ERROR: /home/richard/projects/cert-manager/cert-manager/pkg/util/versionchecker/testdata/BUILD.bazel:9:8: no such package '': error globbing [**] - [bazel-*/**, .git/**]: /home/richard/projects/cert-manager/cert-manager/bazel-cert-manager/external/go_sdk/test (No such file or directory) and referenced by '//pkg/util/versionchecker/testdata:test_manifests'
ERROR: Analysis of target '//pkg/util/versionchecker/testdata:all-srcs' failed; build aborted: Analysis failed
INFO: Elapsed time: 26.190s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (24 packages loaded, 76 targets configured)
FAILED: Build did NOT complete successfully (24 packages loaded, 76 targets configured)

```

Reply:
I re-ran these tests later and they passed :shrug: 

```sh
$ bazel test //pkg/util/...

...

v1.5.2.yaml
v1.5.3.yaml
v1.6.0-alpha.0.41-3fa237cd5b762f-dirty.yaml
v1.6.0-alpha.0.yaml
INFO: Elapsed time: 50.585s, Critical Path: 48.47s
INFO: 84 processes: 10 internal, 74 linux-sandbox.
INFO: Build completed successfully, 84 total actions
//pkg/util:go_default_test                                               PASSED in 0.0s
//pkg/util/cmapichecker:go_default_test                                  PASSED in 0.1s
//pkg/util/pki:go_default_test                                           PASSED in 3.4s
//pkg/util/predicate:go_default_test                                     PASSED in 0.2s
//pkg/util/versionchecker:go_default_test                                PASSED in 13.4s

Executed 5 out of 5 tests: 5 tests pass.
INFO: Build completed successfully, 84 total actions

```

Reply:
I somehow got reading about a tool called `bazelisk` which apparently makes it easier to install and run bazel...anyway, their README says this:

> Bazel does check the .bazelversion file itself, but the failure when it mismatches with the actual version of Bazel can be quite confusing to developers.

https://github.com/bazelbuild/bazelisk#ensuring-that-your-developers-use-bazelisk-rather-than-bazel

So should we also create a `.bazelversion` file ?

Reply:
> ERROR: /home/richard/projects/cert-manager/cert-manager/pkg/util/versionchecker/testdata/BUILD.bazel:9:8: no such package '': error globbing [**] - [bazel-*/**, .git/**]: /home/richard/projects/cert-manager/cert-manager/bazel-cert-manager/external/go_sdk/test (No such file or directory) and referenced by '//pkg/util/versionchecker/testdata:test_manifests'

This may be worth investigating (perhaps separately from this PR). I've seen a similar line in test logs, but never looked into it.

Reply:
> So should we also create a .bazelversion file ?

Bazel would then fail before running `check_min_bazel_version`